### PR TITLE
HFShim: support MPS device

### PR DESCRIPTION
**Pull request by danieldk**
Before this change, two devices and map locations were supported:

- CUDA: `cuda:N`
- CPU: `cpu`

This change adds support for other devices like Metal Performance Shader (MPS) devices by mapping to the active Torch device.

### CODE REVIEW

File: spacy_transformers/layers/hf_shim.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from pathlib import Path
 import srsly
 import torch
-from thinc.api import get_current_ops
+from thinc.util import get_torch_default_device
```
Comment (svlandeg) [inline spacy_transformers/layers/hf_shim.py:6]: Sorry if I missed that conversation, but why is this not accessible from `thinc.api`?

Comment (danieldk) [reply spacy_transformers/layers/hf_shim.py]: Fixed.

**Comment by danieldk** [comment]
I think we also need a dev version of spaCy to get the right Thinc version for this PR.

**Comment by danieldk** [comment]
> Can this be done soon after the v3.4.0 release instead? It's not going to break anything in `spacy-transformers` or the `trf` models, right?

That would work. It shouldn't break anything, just moving models to the right device in a more generic manner.

**Comment by danieldk** [comment]
Set the minimum spaCy version to 3.4.0 and merged in master. This is now ready for review.

**Comment by adrianeboyd** [comment]
@explosion-bot please test_gpu

**Pull request merged**